src/openssl_auth.cpp

```diff
@@ -315,6 +316,7 @@ unsigned char* s3fs_sha256hexsum(int fd, off_t start, ssize_t size)
 
   // seek to top of file.
   if(-1 == lseek(fd, start, SEEK_SET)){
+    EVP_MD_CTX_destroy(sha256ctx);
```

Comment:
Can we sink `sha26ctx` initialization closer to its first use to avoid the first two calls to `EVP_MD_CTX_destroy`?  This will more closely match `nss_auth.cpp:s3fs_sha256hexsum`.